homeassistant/components/accuweather/coordinator.py

```diff
@@ -0,0 +1,80 @@
+"""The AccuWeather coordinator."""
+
+from asyncio import timeout
+from datetime import timedelta
+import logging
+from typing import TYPE_CHECKING, Any
+
+from accuweather import AccuWeather, ApiError, InvalidApiKeyError, RequestsExceededError
+from aiohttp.client_exceptions import ClientConnectorError
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.update_coordinator import (
+    TimestampDataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import DOMAIN, MANUFACTURER
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AccuWeatherDataUpdateCoordinator(TimestampDataUpdateCoordinator[Any]):
+    """Class to manage fetching AccuWeather data API."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        accuweather: AccuWeather,
+        name: str,
+        coordinator_type: str,
+        update_interval: timedelta,
+        update_method: str,
+    ) -> None:
+        """Initialize."""
+        self.accuweather = accuweather
+        self._update_method = update_method
```

Comment:
Why do we make this abstract? We could implement this twice and enjoy the type safety it adds right?

Reply:
Do you mean two separate classes of data update coordinators?